Adventure game tutorial

djswizzy opened the pull request:
Simple adventure game for learners to learn basic if statements

djswizzy committed:
Working game, needs text refinement

djswizzy committed:
Updated version

djswizzy committed:
linting

djswizzy committed:
linting

djswizzy committed:
Added licensing

KywyErik commented on examples/games/Adventure/Adventure.ino:
```diff
@@ -0,0 +1,87 @@
+      onLeftScreen = false;                                                                   //Track that we're not on the left screen
+      onRightScreen = false;                                                                  //Track that we're not on the right screen
+      engine.display.drawText(centerX, centerY - 20, "Go Right");                             // Draw text
+      engine.display.drawText(centerX - 60, centerY - 20, "Go Left");                         // Draw text
```
looks good other than that this text is very hard to read since it's overlaid directly on the bitmap, can we put a box around it or something to make it clearer?

djswizzy replied:
I've addressed these changes

djswizzy committed:
Added text boxes

Joe-Lannan committed:
Linearized the logic flow to make it easier to understand and edit for students, now utlizes a wait for button behavior in the main loop.

Joe-Lannan commented:
I changed the logic a bit so its linear and easier for students to understand.

Joe-Lannan committed:
Improve Adventure game and add comprehensive tutorials

- Enhanced Adventure game structure with helper functions
- Added multi-line text support and improved input handling
- Created 5-part tutorial series for Adventure game development
- Applied clang formatting and code improvements
- Cleaned up Clicker examples (removed test files)

Joe-Lannan committed:
Add bonus tutorial part explaining helper functions

- Added part_6.md with comprehensive explanation of helper functions
- Updated tutorials index to highlight the bonus content
- Covers waitForInput(), drawInstructionText(), and drawBottomText()
- Includes best practices and examples for creating helper functions

Joe-Lannan committed:
Edits to adveture tutorial pages progressing through increasingly advanced control sequences

Joe-Lannan committed:
edit guides/images.md to discuss how to import hpp files

KywyErik commented:
Need to add `- adventure/` to `docs/tutorials/.nav.yaml`, make sure it's in the same order as the tutorials are listed on the main tutorial page

KywyErik commented on docs/tutorials/adventure/part_4.md:
```diff
@@ -0,0 +1,422 @@
+- Harder to see all possible paths
+
+This is completely optional - your simple sequential choices work great too! In the next part, we'll learn how to add custom sprites to make your game look amazing.
+        score += 40;
```
missing some code tags or maybe the first part of the code here

KywyErik commented on docs/tutorials/adventure/part_6.md:
```diff
@@ -0,0 +1,256 @@
+# SPDX-FileCopyrightText: 2025 KOINSLOT, Inc.
```
this needs to be in markdown comment

KywyErik commented on docs/tutorials/adventure/part_6.md:
```diff
@@ -0,0 +1,256 @@
+# SPDX-FileCopyrightText: 2025 KOINSLOT, Inc.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
```
need to copy the html tags from another tutorial last page that hides the "Next" button at the bottom

KywyErik reviewed (changes requested):
Needs a few tweaks to the docs

Joe-Lannan committed:
Fix broken part_5.md for adventure tutorial

Joe-Lannan committed:
fix comments

KywyErik commented on docs/tutorials/bouncing_ball/part_5.md:
```diff
@@ -410,7 +410,7 @@ And there we have it! A Kywy style ball bouncing around the screen and respondin
 
     The full list of events that you can recieve from the Kywy engine is
-    [here](../../../kywy/namespace_kywy_1_1_events/#enum-kywyevents).
+    [here](../../api_reference.md).
```
this should link to the actual events page https://docs.kywy.io/kywy/namespaceKywy_1_1Events/

KywyErik replied:
looks like the link changed or something

Joe-Lannan replied:
it moved but now goes to events.

KywyErik commented on docs/tutorials/index.md:
```diff
@@ -21,6 +21,12 @@ _audience: beginner (middle school age and up)_
 
 Learn programming fundamentals by building a fun Cookie Clicker game! This tutorial teaches variables, input handling, and animations through a simple, engaging project.
 
+### [Choose Your Own Adventure](./adventure/part_1.md)
```
this should be in the same order as the index

Joe-Lannan replied:
Done, ordered roughly by difficulty

Joe-Lannan committed:
Fixed order of tutorials index and nav

Joe-Lannan committed:
Fix missing event enums link in bouncing ball tutorial part 5
